x-pack/plugins/infra/public/components/asset_details/tabs/overview/alerts.tsx

```diff
@@ -91,10 +102,17 @@ export const AlertsSummaryContent = ({
 interface MemoAlertSummaryWidgetProps {
   alertsQuery: AlertsEsQuery;
   dateRange: TimeRange;
+  setIsAlertSectionOpen: (value: EuiAccordionProps['forceState']) => void;
+  setActiveAlertsCount: (value: number) => void;
 }
 
 const MemoAlertSummaryWidget = React.memo(
-  ({ alertsQuery, dateRange }: MemoAlertSummaryWidgetProps) => {
+  ({
+    alertsQuery,
+    dateRange,
+    setIsAlertSectionOpen,
+    setActiveAlertsCount,
```

Comment:
Suggestion: instead of passing the setters, could we pass a callback function that encapsulates these 2 setters? This way, the consumer would decide how to handle the counter event, and leave the memoized `AlertSummaryWidget` component just responsible for rendering the component

eg: 

```tsx
const onAlertsCount = (alertsCount: AlertsCount) => {
   const hasActiveAlerts =
        typeof alertsCount?.activeAlertCount === 'number' && alertsCount?.activeAlertCount > 0;

      setIsAlertSectionOpen(hasActiveAlerts ? 'open' : 'closed');
      setActiveAlertsCount(alertsCount?.activeAlertCount ?? 0);
}

<MemoAlertSummaryWidget
        onAlertsCount={onAlertsCount}
        alertsQuery={alertsEsQueryByStatus}
        dateRange={dateRange}
      />
```

Then here

```tsx
const MemoAlertSummaryWidget = React.memo(
  ({
    alertsQuery,
    dateRange,
    onAlertsCount,
} => {
...
  return (<AlertSummaryWidget
        chartProps={chartProps}
        featureIds={infraAlertFeatureIds}
        filter={alertsQuery}
        timeRange={summaryTimeRange}
        getAlertsCount={onAlertsCount}
        fullSize
        hideChart
      />)
}
```


```

Reply:
Good idea, changed 👍